Build /ai/mlops

mtruj013 opened the pull request:
## Done

- Built new page as per [doc](https://docs.google.com/document/d/18AMX_-3zjBNUNGr0X5QtjKhsUzAc2k8VM1-opZEGh4k/edit) and [design](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?node-id=248%3A1446&mode=dev)

## QA

- View the site locally in your web browser at: https://ubuntu-com-13146.demos.haus/ai/mlops
    - Be sure to test on mobile, tablet and desktop screen sizes
- Check against doc and design

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5512

webteam-app commented:
Demo starting at https://ubuntu-com-13146.demos.haus

lyubomir-popov commented:
this is blurred; can it be an svg?
![image](https://github.com/canonical/ubuntu.com/assets/2741678/5946c8c1-ddd5-454b-8171-81c56aea5266)

also blurred, ideally it should be an svg 
![image](https://github.com/canonical/ubuntu.com/assets/2741678/8ce0155b-ee4b-4fe5-8f1c-525500f7de8f)


this should be a shallow section:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/8b4ff096-8f19-4c2e-947f-4c17885c88a3)

The rest looks perfect, thanks a lot!

juanruitina commented:
Very minor stuff:

- On the H1, can you please place the break after "scale"?
- "Talk to our experts" should be a `p` element.

One last thing: we are hiding the diagrams on mobile, ideally we should keep them given they contain information. First one downscales reasonably well, second one doesn't though. I suggest to show the first one and keep the second hidden so it doesn't become a blocker. But maybe something to consider when/if we enrich @lyubomir-popov?

mtruj013 commented:
@lyubomir-popov @juanruitina thanks both! I've addressed your comments if you could take another look?

Lyubo about the assets, both are already svgs. I adjusted the sizing very slightly and that seemed to help for the second one, the first one looks the same to me regardless (although to be fair, I don't really notice any blurring in the first one: 
![image](https://github.com/canonical/ubuntu.com/assets/17607612/3ee0177d-4747-4761-9ca4-1d016c9414a1)
)

Asset links here if it's helpful:
- https://assets.ubuntu.com/v1/0f2554e7-what%20is%20MLOps%20illustration.svg
- https://assets.ubuntu.com/v1/a59f849e-what%20is%20MLOps%20illustration.svg

lyubomir-popov commented:
says mlops services twice. other than that, looks good to me 
![image](https://github.com/canonical/ubuntu.com/assets/2741678/cccab6fa-7c30-4884-bd3b-f1ec9ff90337)

akbarkz commented on static/sass/_pattern_strip.scss:
```diff
@@ -1023,3 +1024,23 @@
+    background-size: contain;
+
+    // Hide suru on small and medium screens
+    @media screen and (max-width: $breakpoint-small - 1) {
```
do we always subtract 1 from the breakpoints? Or is it better to overwrite the breakpoint variables to be smaller by 1 px?

mtruj013 replied:
This convention comes from the [Vanilla documentation](https://vanillaframework.io/docs/settings/breakpoint-settings) so, ideally, we should always follow that as it's way easier for future maintenance. Updating the breakpoint variable on u.com only would mean that if there's ever a future Vanilla update that touches that variable, the u.com version will be out of sync with base Vanilla and will also need to be updated which can get pretty messy

juanruitina commented:
The logos on the pages listed above certainly look better now, although there are several instances of blurry logos (they aren't blurry on the live website).

Some examples:

In https://ubuntu-com-13131.demos.haus/telco

![Screenshot 2023-09-13 at 13 34 55](https://github.com/canonical/ubuntu.com/assets/1409729/5768d84e-e68b-474f-a992-5aabffd19750)

In https://ubuntu-com-13131.demos.haus/financial-services

![Screenshot 2023-09-13 at 13 37 44](https://github.com/canonical/ubuntu.com/assets/1409729/815ebc5a-6b8e-4335-b233-92d3a0987348)

![Screenshot 2023-09-13 at 13 32 32](https://github.com/canonical/ubuntu.com/assets/1409729/4307a341-ee55-401a-9694-6abecdfdc566)

In https://ubuntu-com-13131.demos.haus/kubernetes

![Screenshot 2023-09-13 at 13 32 15](https://github.com/canonical/ubuntu.com/assets/1409729/289584ae-3bc2-4889-a0af-3fc9c7b8860d)

Can we address this? In any case, I'd defer to Lyubo for reviewing this, as it's for Visual to decide what's good enough.

juanruitina commented:
Page looks good to me, +1 this assuming the issue with the navigation alignment is addressed with the new Vanilla version.

mtruj013 commented:
@juanruitina to clarify, the scope for this PR is strictly the mlops page, anything regarding the meganav or site-wide logo sections is not meant to be covered here

petesfrench commented on templates/ai/mlops.html:
```diff
@@ -0,0 +1,210 @@
+{% extends "ai/_base_ai.html" %}
```
```suggestion
{% extends "ai/base_ai.html" %}
```
Our convention is to name these without a leading underscore(_)

petesfrench commented:
modal looks a little funky on medium and small screens:
![image](https://github.com/canonical/ubuntu.com/assets/58276363/9d46e147-c52d-46e9-8192-64f648f7718c)

petesfrench commented on templates/ai/mlops.html:
```diff
@@ -0,0 +1,210 @@
+          <p>Move from experimentation to production using a modular MLOps platform. <br class="u-hide--small">Accelerate your project delivery and ensure quick return on investment.</p>
+        </div>
+        <a class="p-button--positive" href="/engage/mlops-guide">Read our guide to MLOps</a>
+        <a class="p-button js-invoke-modal" href="#get-in-touch">Talk to our experts</a>
```
```suggestion
        <a class="p-button js-invoke-modal" href="/ai/contact-us">Talk to our experts</a>
```
We can have the href go to `/ai/contact-us` as a fallback in case the user doesn't have JS enabled

petesfrench commented on templates/ai/mlops.html:
```diff
@@ -0,0 +1,210 @@
+      </div>
+      <p>Machine learning operations (MLOps) is like DevOps for machine learning. It is a set of practices that automates machine learning workflows, ensuring scalability, portability and reproducibility.</p>
+      <div>
+        <img src="https://assets.ubuntu.com/v1/0f2554e7-what is MLOps illustration.svg" alt="" style="height: 267px; width: 597px;">
```
Was this the image you spoke about being converted to a PNG when using the image template?

mtruj013 replied:
Yes, Lyubo noticed a difference between the version of the image straight from the asset server and how it appears after it's been sanitized by cloudinary so we agreed to just link the asset directly

petesfrench commented on templates/ai/mlops.html:
```diff
@@ -0,0 +1,210 @@
+  <div class="row">
+    <div class="col-start-large-4 col-9">
+      <p class="p-heading--2">
+        <a class="js-invoke-modal" href="#get-in-touch">Talk to our experts&nbsp;&rsaquo;</a>
```
```suggestion
        <a class="js-invoke-modal" href="/ai/contact-us!>Talk to our experts&nbsp;&rsaquo;</a>
```

petesfrench commented on templates/ai/mlops.html:
```diff
@@ -0,0 +1,210 @@
+      <div class="p-section--shallow">
+        <hr class="p-rule">
+        <div class="p-section--shallow">
+          <h3 class="p-heading--2">Get your team up-to-date <br class="u-hide--small"> an MLOps workshop</h3>
```
```suggestion
          <h3 class="p-heading--2">Get your team up-to-date <br class="u-hide--small">with a MLOps workshop</h3>
```
Missing a word and I think it should be 'a MLOps', will leave a comment in the doc

mtruj013 replied:
@Sophie-32 can confirm, but I think it actually is supposed to be "an MLOps". Reasoning [here](https://www.scribbr.com/commonly-confused-words/a-vs-an/#:~:text=A%20or%20an%20before%20acronyms,-The%20same%20rule&text=Acronyms%20beginning%20with%20a%20consonant,vowel%20sound%20use%20%E2%80%9Can.%E2%80%9D)

petesfrench reviewed (commented):
Looking good, just a couple small changes in the code to be adressed
